state_machine/drivers/pycubedmini/lib/logging.py

```diff
@@ -0,0 +1,187 @@
+
+    # if info.txt does exist, read the existing info.txt
+    if info_filename in listdir("/sd/"):
+        # read the contents and retrieve logfile_count
+        info_file = open(info_filename_dir, "r")
+        info_file_contents = info_file.read()
+        info_file_lines = info_file_contents.split("\r\n")
+        # get logfile_count
+        logfile_count = int(info_file_lines[2])
+        # close the file
+        info_file.close()
+
+    # return logfile_count read from info.txt
+    return logfile_count
+
+
+def write_infotxt(logfile_count):
+    """
+    write info.txt in a certain format based on logfile_count
+    """
+
+    # overwrite the current info file
+    info_file = open(info_filename_dir, "w")
+    # write a timestamp, the current logfile_count
+    info_file.write(f"{time.monotonic()}\r\n")  # line 0
+    info_file.write("logfile_count:\r\n")      # line 1
+    info_file.write(f"{logfile_count}\r\n")     # line 2
+    # close file
+    info_file.close()
+
+
+def rjust(string, length, char):
+    """
+    python string builtin rjust doesn't exist in circuitpython
+    redefining this function to make filename formatting easier
+    """
+    string_length = len(string)
+    if string_length < length:
+        padding = char * (length - string_length)
+        return f"{padding}{string}"
+
+    if string_length == length:
+        return string
+
+    if string_length > length:
+        return string[0:length]
+
+
+def get_logfile_name_dir(logfile_count):
+    logfile_name = get_logfile_name(logfile_count)
+    return f"/sd/logs/{logfile_name}"
+
+
+def get_logfile_name(logfile_count):
+    """
+    format the logfile_name based on logfile_count and return
+    """
+
+    logfile_count_string = rjust(str(logfile_count), 3, '0')
+    return f"log{logfile_count_string}.txt"
+
+
+def new_log(logfile_count):
+    """
+    Create a new log file
+    Write a header and return the new logfile_count
+    """
+
+    # After you fill up 1000 files (~ 1 GB), overwrite old files
+    logfile_count += 1
+    logfile_count %= 1000
+    logfile_name = get_logfile_name(logfile_count)
+    logfile_name_dir = get_logfile_name_dir(logfile_count)
+
+    # if logfile_name doesn't exist, create it
+    if logfile_name not in listdir("/sd/logs/"):
+        # create logfile and write a header
+        logfile = open(logfile_name_dir, "x")
+        logfile.write(f"########## Created: {time.monotonic()} ##########\r\n")
+        logfile.close()
+    # if logfile_name exists, overwrite
+    else:
+        # overwrite existing logfile and write a header
+        logfile = open(logfile_name_dir, "w")
+        logfile.write(f"########## Created: {time.monotonic()} ##########\r\n")
+        logfile.close()
+
+    # print a confirmation message that a new logfile was created
+    write_infotxt(logfile_count)
+
+    return logfile_count
+
+
+def log(msg, max_file_size=max_file_size):
+    """
+    Create/open file and write logs
```

Comment:
I'm really struggling to understand this function, could you please explain to me what it does and put it into a docstring.